scripts/coccinelle/unsigned_assignment.cocci

```diff
@@ -0,0 +1,40 @@
+// Find assignments to unsigned variables and add an 'U' to the value
+// Copyright (C) 2018 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+virtual patch
+virtual report
+
+@r_unsigned@
```

Comment:
add `depends on !(file in "ext")` to all rules to exclude processing
`ext/` directory from processing.

Ref: https://github.com/zephyrproject-rtos/zephyr/commit/817f4b374fba536db2740fdc8965e35a88ac5e2d

Reply:
When I add this, spatch version 1.0.4.deb-3+b4 from Debian testing complains with:
Fatal error: exception Failure("rule name: parse error: \n = File \".../zephyr/scripts/coccinelle/unsigned_assignment.cocci\", line 11, column 25,  charpos = 311\n    around = 'file', whole content = @r_unsigned depends on !(file in \"ext\")@\n")

It seems I need to test if a newer version of coccinelle/spatch works better.

Reply:
Hmm.
Well, 1.0.4 is too old and we always recommend to use latest version:
```
himanshu@himanshu-Vostro-3559:~/coccinelle$ git diff --shortstat master 1.0.4 
 829 files changed, 18373 insertions(+), 67399 deletions(-)
```
I updated all scripts to use `depends on !(file in "ext")` since `ext/` comes from 
third paty vendor repo and there was less interest for those reports.

And it works fine for me.